### PR TITLE
nettype: introduce type for ng_netif_hdr_t

### PR DESCRIPTION
**Pull request by authmillenon**
Introduces nettype for the `ng_netif_hdr_t` header.

### CODE REVIEW

**Comment by haukepetersen** [comment]
How about putting `NG_NETTYPE_NETIF` in front of the enum and assigning it to `-1`. No need to make threads register to it and we can save the 4 bytes in the netreg.

**Comment by authmillenon** [comment]
fine, reverted e4b01c7.

File: sys/include/net/ng_nettype.h [hunk 1]
```diff
@@ -37,6 +37,11 @@ extern "C" {
  * @note    Expand at will.
  */
 typedef enum {
+    /**
+     * @brief   Protocol is as defined in @ref ng_netif_hdr_t. Not usable with
+     *          @ref net_ng_netreg
```
Comment (haukepetersen) [inline sys/include/net/ng_nettype.h]: `net_ng_netreg`?

Comment (authmillenon) [inline sys/include/net/ng_nettype.h]: Yes, the `ng_netreg` module. You can't register to type `NG_NETTYPE_NETIF` there.

Comment (haukepetersen) [inline sys/include/net/ng_nettype.h]: oh dammit, of course... was slightly confused with the@ref here... never mind.

**Comment by haukepetersen** [comment]
ACK

**Comment by authmillenon** [comment]
Squashed.

**Comment by authmillenon** [comment]
Travis fails because there it can't rebase to master. Merge?

**Comment by authmillenon** [comment]
The rebase was not possible due to reachability issues to github. Now everything is alright.

=> Go.

**Pull request merged**